### ISSUE TITLE
Not working on Resharper 2020.2 EAP 8

### ISSUE THREAD

**Issue by qlikTERror** [MrJul/ReSharper.EnhancedTooltip#160]

**Comment by NN---** [comment]
```
---------------------------
ReSharper – Runtime Error – Technical Data
---------------------------
JetBrains ReSharper has encountered a runtime error.
Technical data follows.
Hint: Ctrl+C copies the text of system message boxes to Clipboard.
________________________________________________________________

Error resolving type JetBrains.Lifetimes.Lifetime from assembly “JetBrains.Platform.RdCore, Version=777.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325”, request originating from assembly “GammaJul.ReSharper.EnhancedTooltip, Version=3.9.0.0, Culture=neutral, PublicKeyToken=null”. Could not find a referenced assembly by the TT_ASSEMBLYREF token.



--- EXCEPTION #1/2 [TypeLoadException]

Message = “Error resolving type JetBrains.Lifetimes.Lifetime from assembly “JetBrains.Platform.RdCore, Version=777.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325”, request originating from assembly “GammaJul.ReSharper.EnhancedTooltip, Version=3.9.0.0, Culture=neutral, PublicKeyToken=null”. Could not find a referenced assembly by the TT_ASSEMBLYREF token.”

ExceptionPath = Root.InnerException

ClassName = System.TypeLoadException

Data.OriginatorAssembly = “MetadataAssembly(GammaJul.ReSharper.EnhancedTooltip, Version=3.9.0.0, Culture=neutral, PublicKeyToken=null)”

Data.RawMessage = “Could not find a referenced assembly by the TT_ASSEMBLYREF token.”

Data.TypeFqn = JetBrains.Lifetimes.Lifetime

Data.AssemblyTypeQualification = “JetBrains.Platform.RdCore, Version=777.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325”

HResult = COR_E_TYPELOAD=80131522

TypeLoadResourceID = 0



--- Outer ---



--- EXCEPTION #2/2 [LoggerException]

Message = “Error resolving type JetBrains.Lifetimes.Lifetime from assembly “JetBrains.Platform.RdCore, Version=777.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325”, request originating from assembly “GammaJul.ReSharper.EnhancedTooltip, Version=3.9.0.0, Culture=neutral, PublicKeyToken=null”. Could not find a referenced assembly by the TT_ASSEMBLYREF token.”

ExceptionPath = Root

ClassName = JetBrains.Util.LoggerException

Data.ManagedThreadName = <NULL>

InnerException = “Exception #1 at Root.InnerException”

HResult = COR_E_APPLICATION=80131600

StackTraceString = “

  at JetBrains.Util.OnError.<>c__DisplayClass17_0.<LogForeignException>b__0(Exception ex)

     at JetBrains.Util.OnError.Handle(Exception ex)

     at JetBrains.Application.Bindings.AssemblyBinderExtensions.<>c__DisplayClass4_0.<CreateMetadataLoader>b__2(UnresolvedTypeArgs args)

     at JetBrains.DataFlow.Signal`1.NotifySinks(TValue payload)

     at JetBrains.DataFlow.Signal`1.Fire(TValue value, Object cookie)

     at JetBrains.DataFlow.Signal`1.Fire(TValue value)

     at JetBrains.Metadata.Reader.Impl.MetadataAssembly.GetTypeInfoFromToken(MetadataToken token)

     at JetBrains.Metadata.Reader.Impl.MetadataAssembly.GetTypeFromToken(MetadataToken token, TypeDecodeContext context)

     at JetBrains.Metadata.Reader.Impl.MetadataAssembly.GetTypeFromToken(MetadataToken token)

     at JetBrains.Metadata.Reader.Impl.MetadataTypeBuilder.BuildClassType(MetadataToken typeDefOrRefOrSpecToken)

     at JetBrains.Metadata.Reader.Impl.TypeBlobDecoder.DecodeTypeDefOrRef[TType,TMethodSig](IBinaryReader reader, ITypeBuilder`2 typeBuilder)

     at JetBrains.Metadata.Reader.Impl.TypeBlobDecoder.DecodeType[TType,TMethodSig](IBinaryReader reader, ITypeBuilder`2 typeBuilder)

     at JetBrains.Metadata.Reader.Impl.MethodSignatureBlobDecoder.DecodeBlob[TType,TSignature](IBinaryReader reader, IMethodSignatureBuilder`2 builder)

     at JetBrains.Metadata.Reader.Impl.MetadataMethod.get_Signature()

     at JetBrains.Metadata.Reader.Impl.MetadataMethod.get_ReturnValue()

     at JetBrains.Application.Table.Emit.EmitCatalogTables.Types.GetTypeDeclarationMembers(TypeRef typeref, IMetadataTypeInfo type, EmitTimeTables tables)

     at JetBrains.Application.Table.Emit.EmitCatalogTables.Types.FillPreCreatedShallowTypeDeclaration(TypeRef typeref, IMetadataTypeInfo type, EmitTimeTables tables)

     at JetBrains.Application.Table.Emit.EmitCatalogTables.Types.FillPreCreatedShallowTypes(EmitTimeTables tables)

     at JetBrains.Application.Table.Emit.EmitCatalogTables.<>c__DisplayClass1_0.<CreateCatalogTablesWithMetadataReader>b__0(Lifetime lifetime)

     at JetBrains.Lifetimes.Lifetime.Using(Action`1 action)

     at JetBrains.Application.Table.Emit.EmitCatalogTables.CreateCatalogTablesWithMetadataReader(IEnumerable`1 assemblies, IAssemblyMasterBinder masterbinder, ICollection`1 traitcalc, Stream output)

     at JetBrains.Application.BuildScript.Application.Catalogs.PartCatalogs.CreateWithMetadataReaderRaw(Lifetime lifetime, IEnumerable`1 assemblies, IAssemblyMasterBinder masterbinder)

     at JetBrains.Application.Environment.DeployedPackagesLoader.CreateAdHocCatalogForUncatalogedAssemblies(Lifetime lifeCreationProcess, ICollection`1 assembliesForAdHocCataloging, DeployedPackagesMasterBinderData binderlist, ICollection`1 catables)

     at JetBrains.Application.Environment.DeployedPackagesLoader.Load(Lifetime lifetime, DeployedPackagesIdentity identity, Lazy`1 taskerForNugets, Diagnostyx diag)

     at JetBrains.Application.Environment.DeployedPackagesCache.LoadWithCaching(Lifetime lifetime, DeployedPackagesIdentity identity, Lazy`1 taskerForNugets)

     at JetBrains.Application.Environment.DeployedPackages.LoadWithCaching(Lifetime lifetime, DeployedPackagesIdentity identity, Lazy`1 taskerForNugets)

     at JetBrains.Application.Environment.HostParameters.CollectProductPackagesInDirectoryFlatHostMixin.JetBrains.Application.Environment.HostParameters.IHaveFullPartCatalogSetHostMixin.GetFullPartCatalogSet(Lifetime lifetime)

     at JetBrains.Application.Environment.HostParameters.JetHostParametersCaller.GetFullPartCatalogSet(Lifetime lifetime)

     at JetBrains.Application.Environment.JetEnvironment..ctor(Lifetime lifetime, JetHostParametersCaller host)

     at JetBrains.Application.Environment.JetEnvironment.<>c__DisplayClass8_0.<InternalCreate>b__0()

     at JetBrains.Threading.ReentrancyGuard.Execute(String name, Action action)

     at JetBrains.Application.Environment.JetEnvironment.InternalCreate(LifetimeDefinition lifetimeDefinition, JetHostParametersCaller host, JetEnvironment& environment)

     at JetBrains.Application.Environment.JetEnvironment.<>c__DisplayClass6_0.<CreateDontRun>b__0()

     at JetBrains.Lifetimes.LifetimeDefinition.Bracket(Action opening, Action closing)

     at JetBrains.Application.Environment.JetEnvironment.CreateDontRun(Lifetime lifetime, JetHostParameters hostparams)

     at JetBrains.VsIntegration.Shell.Package.VsPackage.Init_CreateHostAndEnvironment(Lifetime lifetime, IServiceProvider site)

     at JetBrains.VsIntegration.Shell.Package.VsPackage.<.ctor>b__3_0(Lifetime lifeHaveSite, IServiceProvider site)

     at JetBrains.DataFlow.IPropertyEx.<>c__DisplayClass29_0`1.<ForEachValue>b__0(Lifetime lf, TValue value, Object cookie)

     at JetBrains.DataFlow.IPropertyEx.<>c__DisplayClass30_1`1.<ForEachValue>b__1(Lifetime lifetimeValue)

     at JetBrains.Lifetimes.SequentialLifetimes.<>c__DisplayClass4_0.<Next>b__0(LifetimeDefinition lifetimeDefinition)

     at JetBrains.Lifetimes.LifetimeDefinition.ExecuteOrTerminateOnFail(Action`1 atomicAction)

     at JetBrains.Lifetimes.SequentialLifetimes.<>c__DisplayClass5_0.<DefineNext>b__0(LifetimeDefinition definition)

     at JetBrains.Lifetimes.SequentialLifetimes.TrySetNewAndTerminateOld(LifetimeDefinition newLifetimeDefinition, Action`1 actionWithNewLifetime)

     at JetBrains.Lifetimes.SequentialLifetimes.DefineNext(Action`1 atomicAction)

     at JetBrains.Lifetimes.SequentialLifetimes.Next(Action`1 atomicAction)

     at JetBrains.DataFlow.IPropertyEx.<>c__DisplayClass30_0`1.<ForEachValue>b__0(PropertyChangedEventArgs`1 args)

     at JetBrains.DataFlow.Signal`1.NotifySinks(TValue payload)

     at JetBrains.DataFlow.Signal`1.Fire(TValue value, Object cookie)

     at JetBrains.DataFlow.Property`1.FireChange(TValue old, TValue new, Object cookie)

     at JetBrains.DataFlow.Property`1.SetValue(TValue value, Object cookie)

     at JetBrains.DataFlow.Property`1.set_Value(TValue value)

     at JetBrains.VsIntegration.Shell.Package.VsPackageSdkStub.<.ctor>b__11_5()

     at JetBrains.Threading.JetDispatcher.Closure.Execute()

     at JetBrains.Util.Concurrency.WinJetDispatcher.ProcessQueue(Int32 nMinBucket)

     at JetBrains.Util.Concurrency.WinJetDispatcher.ExternalIdleProcessing.OnExternalDriverIdle()

     at JetBrains.VsIntegration.Shell.VsDoIdleDispatcherSink.Microsoft.VisualStudio.OLE.Interop.IOleComponent.FDoIdle(UInt32 grfidlef)

”


---------------------------
OK   
---------------------------

```

**Comment by pinkfloydx33** [comment]
Get similar issue with Resharper 2020.2 (non-EAP):

> Install completed with 3 errors.
  05:25:48.149 ReSharperPlatformVs16_17035c09: Error resolving type JetBrains.Lifetimes.Lifetime from assembly “JetBrains.Platform.RdCore, Version=777.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325”, request originating from assembly “GammaJul.ReSharper.EnhancedTooltip, Version=3.9.0.0, Culture=neutral, PublicKeyToken=null”. Could not find a referenced assembly by the TT_ASSEMBLYREF token.
  05:25:48.211 ReSharperPlatformVs16_17035c09: Error resolving type JetBrains.Lifetimes.Lifetime from assembly “JetBrains.Platform.RdCore, Version=777.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325”, request originating from assembly “GammaJul.ReSharper.EnhancedTooltip, Version=3.9.0.0, Culture=neutral, PublicKeyToken=null”. Could not find a referenced assembly by the TT_ASSEMBLYREF token.
  05:25:48.221 ReSharperPlatformVs16_17035c09: Error resolving type JetBrains.Lifetimes.Lifetime from assembly “JetBrains.Platform.RdCore, Version=777.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325”, request originating from assembly “GammaJul.ReSharper.EnhancedTooltip, Version=3.9.0.0, Culture=neutral, PublicKeyToken=null”. Could not find a referenced assembly by the TT_ASSEMBLYREF token.

**Comment by DimitarCC** [comment]
Well it have to be rebuilt with lates SDK so to work with 2020.2. I have succesfully rebuilt it with some adjustments of the code.

**Comment by ContentsMayVary** [comment]
With the released version of Resharper 2020.2, the EnhancedTooltip extension doesn't even show up in the Extension Manager (which is fair enough, since EnhancedTooltip doesn't currently work with that verison of Resharper)

**Comment by pinkfloydx33** [comment]
It's weird, it doesn't show up on my work machine but it does on my personal computer. That's how I got the errors. I've not been able to determine what the difference between them is though.

**Comment by ContentsMayVary** [comment]
@pinkfloydx33 It's now showing up for my PC again, maybe because I've rebooted since yesterday? (The version showing up is 3.9.0-eap04.)

**Comment by DimitarCC** [comment]
@ContentsMayVary 3.9.0-eap04 is something very old. 3.13 is the latest so maybe thats why there is errors. As i said before it needs to be rebuilt with Resharper SDK 2020.2.0 and Wave 202. Then it will work and show properly in released Resharper 2020.2

**Comment by MrJul** [comment]
Enhanced Tooltip v3.14 for ReSharper 2020.2 RTM has been released.
It'll be available in the extension manager as soon as JetBrains approves it.
